pop-miners/altchain-pop-miner/src/main/kotlin/org/veriblock/lite/transactionmonitor/TransactionMonitor.kt

```diff
@@ -43,122 +49,64 @@ class TransactionMonitor(
         }
     }
 
-    fun getTransactions(): Collection<WalletTransaction> =
-        Collections.unmodifiableCollection(transactions.values)
-
-    private fun save() {
-        val diskWallet = File(context.directory, context.filePrefix + TM_FILE_EXTENSION)
-
-        lock.withLock {
-            try {
-                FileOutputStream(diskWallet).use { stream ->
-                    stream.writeTransactionMonitor(this@TransactionMonitor)
+    fun start() {
+        GlobalScope.launch {
+            while (isActive) {
+                delay(Duration.ofSeconds(60).toMillis())
+                try {
+                    checkPendingTransactions()
+                } catch (e: Exception) {
+                    logger.error { e.message; e }
```

Comment:
```suggestion
                    logger.error(e) { e.message; }
```
You can pass the 'e' like that, also the message should be a little more descriptive/graceful than the exception message